Lib/tkinter/test/test_tkinter/test_misc.py

```diff
@@ -192,6 +192,51 @@ def test_clipboard_astral(self):
         with self.assertRaises(tkinter.TclError):
             root.clipboard_get()
 
+    def test_mainloop_dispatching(self):
+        # reconstruct default root destroyed by AbstractTkTest
+        root = tkinter._default_root = self.root
+        for thing in (root.tk, root, tkinter):
+            #mainloop, dispatching, willdispatch
+            self.assertFalse(thing.dispatching())
+            root.after(0, lambda:self.assertTrue(thing.dispatching()))
+            root.after(0, root.quit)
```

Comment:
Tcl provides as close to a guarantee of this as you can get with the `after` command.  Do you mean that we should make sure the test doesn't hang in the face of future bugs?

Reply:
Running `root.after(0, print, "tst")` in the console immediately prints "tst" even though the mainloop hasn't been called. I haven't investigated the cause (no, I'm not using IDLE) but it seems it doesn't necessarily wait for the mainloop to be called.

Reply:
That is an artifact of the REPL EventHook. It is [deep magic](https://stackoverflow.com/questions/20891710/what-magic-prevents-tkinter-programs-from-blocking-in-interactive-shell). It should not be an issue in a script.

Reply:
This brings a question to mind, should EventHook set the dispatching flag as well? We should at least document a clear cut distinction between REPL behavior and script behavior somewhere.

Reply:
I believe it would make sense to set dispatching in EventHook. Internally, I don't think it makes a difference but for users who call the `dispatching` method it could be quite useful (so long as we properly document this behaviour to avoid confusion). And yes, we should definitely document the REPL behaviour (though this should probably be a different issue & PR)!

Reply:
Then I propose we leave the return value of `dispatching` as PyLong rather than PyBool. For this PR it will be a one-line change, but for the future we can its value depending on what is dispatching and map that to an IntEnum for REPL, mainloop, ipython, and whoever else.